registration/views/printing.py

```diff
@@ -16,16 +31,139 @@ def printNametag(request):
 
 
 def servePDF(request):
-    filename = request.GET.get("file", None)
-    if filename is None:
-        return JsonResponse({"success": False, "reason": "Bad request"}, status=400)
-    filename = filename.replace("..", "/")
+    if getattr(settings, "PRINT_RENDERER", "wkhtmltopdf") == "gotenberg":
+        return pdfFromGotenberg(request)
+    else:
+        return pdfFromDisk(request)
+
+
+def pdfFromDisk(request: HttpRequest) -> Union[FileResponse, JsonResponse]:
+    name = request.GET.get("file", None)
+    if not name or not isinstance(name, str):
+        return JsonResponse(
+            {"success": False, "reason": "Name was missing"}, status=400
+        )
+
+    root_dir = getattr(settings, "PDF_DIRECTORY", "/tmp")
     try:
-        fsock = open(f"/tmp/{filename}", "rb")
-    except IOError as e:
-        return JsonResponse({"success": False, "reason": "file not found"}, status=404)
-    response = FileResponse(fsock, content_type="application/pdf")
-    # response['Content-Disposition'] = 'attachment; filename=download.pdf'
+        path = Path(root_dir).joinpath(Path(name).name)
+        f = open(path, "rb")
+    except IOError:
+        return JsonResponse({"success": False, "reason": "IO error"}, status=404)
+
+    response = FileResponse(f, content_type="application/pdf")
     response["Access-Control-Allow-Origin"] = "*"
-    # os.unlink("/tmp/%s" % filename)
     return response
+
+
+def pdfFromGotenberg(request: HttpRequest) -> Union[HttpResponse, JsonResponse]:
```

Comment:
Would you mind adding a test to breeze through the happy path for this new function?  :pray: 

https://coveralls.io/builds/70853650/source?filename=registration%2Fviews%2Fprinting.py#L34